runtime/lua/vim/_editor.lua

```diff
@@ -686,53 +703,46 @@ function vim._cs_remote(rcid, server_addr, connect_error, args)
     return "E247: " .. explanation .. ". " .. consequence
   end
 
-  local f_silent = false
-  local f_tab = false
-
-  local subcmd = string.sub(args[1],10)
-  if subcmd == 'tab' then
-    f_tab = true
-  elseif subcmd == 'silent' then
-    f_silent = true
-  elseif subcmd == 'wait' or subcmd == 'wait-silent' or subcmd == 'tab-wait' or subcmd == 'tab-wait-silent' then
-    return { errmsg = 'E5600: Wait commands not yet implemented in nvim' }
-  elseif subcmd == 'tab-silent' then
-    f_tab = true
-    f_silent = true
-  elseif subcmd == 'send' then
+  local flags = {}
+  local possible = { tab = true, wait = true, silent = true, send = true, expr = true }
+  for subcommand in string.gmatch(string.sub(args[1], #'--remote' + 1), '-(%a+)') do
+    if not possible[subcommand] then
+      return { errmsg='Unknown option argument: ' .. args[1] }
+    end
+    flags[subcommand] = true
+    if subcommand == 'tab' then
+      possible = { wait = true, silent = true }
+    elseif subcommand == 'wait' then
+      possible = { silent = true }
+    else -- we hit silent, send, or expr. Loop again to make sure we didn't get another subcommand
+      possible = {}
+    end
+  end
+
+  if flags['send'] then
     if rcid == 0 then
       return { errmsg = connection_failure_errmsg('Send failed.') }
     end
     vim.fn.rpcrequest(rcid, 'nvim_input', args[2])
-    return { should_exit = true, tabbed = false }
-  elseif subcmd == 'expr' then
+  elseif flags['expr'] then
     if rcid == 0 then
       return { errmsg = connection_failure_errmsg('Send expression failed.') }
     end
     print(vim.fn.rpcrequest(rcid, 'nvim_eval', args[2]))
-    return { should_exit = true, tabbed = false }
-  elseif subcmd ~= '' then
-    return { errmsg='Unknown option argument: ' .. args[1] }
-  end
-
-  if rcid == 0 then
-    if not f_silent then
+  elseif rcid == 0 then
+    if not flags['silent'] then
       vim.notify(connection_failure_errmsg("Editing locally"), vim.log.levels.WARN)
     end
   else
-    local command = {}
-    if f_tab then table.insert(command, 'tab') end
-    table.insert(command, 'drop')
+    local to_edit = {}
     for i = 2, #args do
-      table.insert(command, vim.fn.fnameescape(args[i]))
+      table.insert(to_edit, vim.fn.fnameescape(args[i]))
     end
-    vim.fn.rpcrequest(rcid, 'nvim_command', table.concat(command, ' '))
+    local remote_chan_id = vim.fn.rpcrequest(rcid, 'nvim_get_api_info')[1]
```

Comment:
This is a silly way to get the channel id for the remote side to callback with. I couldn't see a way to do that from the remote side though. Is there a way from inside an RPC to get the channel id for the other side of the connection? If not, would it make sense for me to add that?